### DISCUSSION TITLE
Access component attributes within css``

### DISCUSSION THREAD

**Discussion by Patrickroelofs** [microsoft/fast#6936] [Q&A] [answered]
I was wondering if there was a way to access attributes within the CSS and dynamically set them based on properties, like for example changing the colour of an element when the isLoading attribute is set to true. This would be similar to how you can use attributes within the template html`` function.

_(example of what im trying to do)_
```ts
export const dataGridStyles = () => css`

    :host {
        ${x => x.isLoading && css`
              color: red;
        `}
     }

`
```

I know the CSS function also has the following attributes, but i have been unable to access the component attributes through that.
```ts
context: ElementDefinitionContext,
definition: FoundationElementDefinition,
 ```

Is this possible or should i look for alternative options?

PS: I do know that we can do `:host([is-loading='true']) { }`, although I would prefer to do it with the attributes in the component itself.

**Comment by DavidVollmers** [comment, marked as answer]
You can use the `PropertyStyleSheetBehavior` class for this. (https://www.fast.design/docs/api/fast-foundation.propertystylesheetbehavior/)

```js
export const myStyles = css`
  /* default styles */
`.withBehaviors(
  new PropertyStyleSheetBehavior('isLoading', true, css`
    /* loading styles */
  `)
);
```

**Reply by Patrickroelofs** [reply]
Thanks, exactly what I was looking for!
